validations/src/validations.rs

```diff
@@ -1677,6 +1754,82 @@ pub fn validate_block_transactions(
         );
     }
 
+    // validate stake transactions in a block
+    let mut st_mt = ProgressiveMerkleTree::sha256();
+    let mut st_weight: u32 = 0;
+
+    // Check if the block contains more than one stake tx from the same operator
+    for i in 1..block.txns.stake_txns.len() {
+        let found = block.txns.stake_txns[i..].iter().find(|stake_tx| {
+            let stake_tx_aux = &block.txns.stake_txns[i - 1];
+
+            stake_tx.body.output.authorization.public_key
+                == stake_tx_aux.body.output.authorization.public_key
+        });
+
+        // TODO: refactor
+        if found.is_some() {
+            return Err(BlockError::RepeatedStakeOperator {
+                pkh: found
+                    .unwrap()
+                    .body
+                    .output
+                    .authorization
+                    .public_key
+                    .pkh()
+                    .hash(),
+            }
+            .into());
+        }
+    }
+
+    for transaction in &block.txns.stake_txns {
+        let (inputs, _output, fee, weight, change) = validate_stake_transaction(
+            transaction,
+            &utxo_diff,
+            epoch,
+            epoch_constants,
+            signatures_to_verify,
+        )?;
+
+        total_fee += fee;
+
+        // Update st weight
+        let acc_weight = st_weight.saturating_add(weight);
+        if acc_weight > MAX_STAKE_BLOCK_WEIGHT {
+            return Err(BlockError::TotalStakeWeightLimitExceeded {
+                weight: acc_weight,
+                max_weight: MAX_STAKE_BLOCK_WEIGHT,
+            }
+            .into());
+        }
+        st_weight = acc_weight;
+
+        // TODO: refactor
+        if change.is_some() {
+            let mut outputs: Vec<&ValueTransferOutput> = Vec::new();
+            let val = change.clone().unwrap();
+            outputs.push(&val);
+            update_utxo_diff(&mut utxo_diff, inputs, outputs, transaction.hash());
+        } else {
+            update_utxo_diff(&mut utxo_diff, inputs, Vec::new(), transaction.hash());
+        }
```

Comment:
```suggestion
        let outputs = change.into_iter().collect::<Vec<_>();
        update_utxo_diff(&mut utxo_diff, inputs, outputs, transaction.hash());
```

If the borrow checker doesn't like `.into_iter()` , then try with `.cloned().iter()`.